### DISCUSSION TITLE
Can't add more spacing types while configuring theme.

### DISCUSSION THREAD

**Discussion by viktapas** [mantinedev/mantine#2288] [Q&A]
Hi there,

We are trying to use this promising UI library and out of the box Mantine expects spacing to be of following type only.  
`export declare type MantineSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';`. 
Since we are having more spacing sizes (`xxl, xxs`), it there a way to extends string literal type `MantineSize` or any other solution?  

Thank you.

**Comment by rtivital** [comment]
You can try overriding MantineSize type the same way as colors – https://mantine.dev/theming/colors/#add-custom-colors-types

**Reply by lavishmantri** [reply]
Hi @rtivital, Is it possible to give an example of this? Since for colors, there is a override option provided, but for sizes there is none.

**Reply by rtivital** [reply]
No, I do not have any working examples for this, you can try doing it on your side and sharing the results.

**Reply by goszczynskip** [reply]
I believe this will be impossible to do with just plain module augmentation since MantineSize is defined as a type. After looking into the source code the theme sizes definitions lack an interface to override. 

The only solution I've found so far is to use package patching eg. via yarn.

**Reply by Azzerty23** [reply]
@rtivital can I open a PR for this feature ? 

I would like to have your vision before, to not waste your time.
Like colors with `MantineThemeColorsOverride`, do you agree to override each "item" in `MantineTheme` interface that use `MantineSize` in a similar fashion ? :
- fontSizes (with `MantineThemeFontSizesOverride`)
- radius (with `MantineThemeRadiusOverride`)
- breakpoints (with `MantineThemeBreakpointsOverride`)
- shadows  (with `MantineThemeShadowsOverride`)

I already made an implementation to have something working especially for breakpoints but I think it's too coupled : I overrided `ThemeSize` directly with a `MantineThemeSizesOverride` but as it's used "everywhere" I had to change `MantineSize` into `DefaultMantineSize` in all components to be sure to not break anything... : [see all changes here](https://github.com/Azzerty23/mantine/pull/1) / [MantineSize.ts](https://github.com/Azzerty23/mantine/pull/1/files#diff-c879de79ebc72b92a2b4dfc8787dbbd6ff80963b4acae08894aafa6cfa6ba927) / [MantineTheme.ts](https://github.com/Azzerty23/mantine/pull/1/files#diff-a996c94ed2bed5e21fa58e9f82bea0b1bc53f4da42ad3260b83a8c0beb523a94)

**Reply by rtivital** [reply]
Currently, it is out of the scope, it can be added after 6.0 is released

**Reply by Azzerty23** [reply]
Noted, I'll be patient
